tests/integration/test_requirements_validation.py

```
@@ -0,0 +1,209 @@
+        """
+        Ensure requirements-dev.txt contains no duplicate package entries.
+        
+        This test treats each non-empty, non-comment line as a package specification and compares
+        package names case-insensitively while ignoring common version specifiers, asserting
+        that no package appears more than once.
+        
+        Parameters:
+            requirements_dev_content (str): Contents of requirements-dev.txt.
+        """
+        import re
+        lines = [l.strip() for l in requirements_dev_content.split('\n') 
```

Ruff (0.14.8)

Ambiguous variable name: `l`

(E741)